irods/files/rule-bases/ipc-housekeeping.re

```diff
@@ -130,16 +130,54 @@ ipc_rescheduleStorageFreeSpaceDetermination {
 
 _ipc_rmTrash {
   writeLine('serverLog', 'DS: starting trash removal');
+  *zone = ipc_ZONE;
+  *verdict = true;
+  msiGetSystemTime(*timestamp, "");
+
+  # 2,592,000 is the number of seconds in 30 days, items older than one month in the trash will be deleted.
+  *month_timestamp = int(*timestamp) - 2592000;
+
+  *KeyValStr = "";
+  msiAddKeyValToMspStr("irodsAdminRmTrash", "", *KeyValStr);
+  foreach(*Row in SELECT META_COLL_ATTR_VALUE, COLL_NAME
+                    WHERE COLL_NAME like '/*zone/trash/%'
+                      AND META_COLL_ATTR_NAME = 'ipc::trash_timestamp'
+                        AND META_COLL_ATTR_VALUE <= month_timestamp) {
+                          *rowCollName = *Row.COLL_NAME;
+                          *status = errorcode(msiRmColl(*rowCollName, *KeyValStr, *Status));
+                          if (*status == 0) {
+                            writeLine("serverLog", "DS: Removed trash collection - *rowCollName");
+                          } else {
+                            writeLine("serverLog", "DS: Unable to remove trash collection - *rowCollName, error code returned *status");
+                            *verdict = false;
+                          }
+  }
 
-  msiGetFormattedSystemTime(*date, 'human', '%d-%02d-%02d');
-  *logFileArg = execCmdArg("/var/lib/irods/log/trash-removal.log-*date");
+  foreach(*Row in SELECT META_DATA_ATTR_VALUE, DATA_NAME, COLL_NAME
+                    WHERE COLL_NAME like '/*zone/trash/%'
+                      AND META_DATA_ATTR_NAME = 'ipc::trash_timestamp'
+                        AND META_DATA_ATTR_VALUE <= month_timestamp) {
+                          *KeyValStr = "";
```

Comment:
`month_timestamp` is a variable, so it needs to be referred to as `*month_timestamp` to be syntactically correct.

Reply:
It also needs to be wrapped in single quotes, i.e. `'*month_timestamp'`.

Reply:
Thank you!